ui/src/app/pages/data/criteria-Search.tsx

```diff
@@ -0,0 +1,234 @@
+    height: 1rem;
+    width: 6.4rem;
+    line-height: 0.7rem;
+  }
+  .p-growl .p-growl-item-container .p-growl-item .p-growl-image {
+    font-size: 1rem !important;
+    margin-top: 0.19rem
+  }
+  .p-growl-item-container:after {
+    content:"";
+    position: absolute;
+    left: 97.5%;
+    top: 0.1rem;
+    width: 0px;
+    height: 0px;
+    border-top: 0.5rem solid transparent;
+    border-left: 0.5rem solid ` + colorWithWhiteness(colors.success, 0.6) + `;
+    border-bottom: 0.5rem solid transparent;
+  }
+  .p-growl-item-container {
+    background-color: ` + colorWithWhiteness(colors.success, 0.6) + `!important;
+  }
+  .p-growl-item {
+    padding: 0rem !important;
+    background-color: ` + colorWithWhiteness(colors.success, 0.6) + `!important;
+    margin-left: 0.3rem;
+  }
+  .p-growl-message {
+    margin-left: 0.5em
+  }
+  .p-growl-details {
+    margin-top: 0.1rem;
+  }
+ `;
+
+interface Props {
+  cohortContext: any;
+  source: string;
+  showHierarchy?: Function;
+}
+
+interface State {
+  backMode: string;
+  autocompleteSelection: Criteria;
+  growlVisible: boolean;
+  groupSelections: Array<number>;
+  hierarchyNode: Criteria;
+  mode: string;
+  selections: Array<Selection>;
+  selectedCriteriaList: Array<any>;
+  selectedIds: Array<string>;
+  treeSearchTerms: string;
+  loadingSubtree: boolean;
+
+}
+export class CriteriaSearch extends React.Component<Props, State>  {
+  growl: any;
+  growlTimer: NodeJS.Timer;
+  subscription: Subscription;
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      autocompleteSelection: undefined,
+      backMode: 'list',
+      growlVisible: false,
+      hierarchyNode: undefined,
+      groupSelections: [],
+      mode: 'list',
+      selectedIds: [],
+      selections: [],
+      selectedCriteriaList: [],
+      treeSearchTerms: '',
+      loadingSubtree: false
+    };
+  }
+
+  componentDidMount(): void {
+    const {cohortContext: {domain, standard, type}} = this.props;
+    let {backMode, mode} = this.state;
+    let hierarchyNode;
+    if (this.initTree) {
+      hierarchyNode = {
+        domainId: domain,
+        type: type,
+        isStandard: standard,
+        id: 0,
+      };
+      backMode = 'tree';
+      mode = 'tree';
+    }
+    this.setState({backMode, hierarchyNode, mode});
+  }
+
+  get initTree() {
+    const {cohortContext: {domain}} = this.props;
+    return domain === DomainType.PHYSICALMEASUREMENT
+        || domain === DomainType.SURVEY
+        || domain === DomainType.VISIT;
+  }
+
+  get isConcept() {
+    return this.props.source === 'concept';
+  }
+
+  getGrowlStyle() {
+    return !this.isConcept ? styles.growl : {...styles.growl, marginRight: '2.5rem', paddingTop: '2.75rem' };
+  }
+
+  searchContentStyle(mode: string) {
+    let style = {
+      display: 'none',
+      flex: 1,
+      minWidth: '14rem',
+      overflowY: 'auto',
+      overflowX: 'hidden',
+      width: '100%',
+      height: '100%',
```

Comment:
Maybe add a `minHeight` here so the Finish and Review button isn't right at the top before results are shown.

![Screen Shot 2020-10-02 at 11 08 15 AM](https://user-images.githubusercontent.com/40036095/94945207-f021f300-049f-11eb-87e4-417c091e0b53.png)

Reply:
Done let me know if it looks better